addons/account_followup/account_followup.py

```diff
@@ -1,125 +1,90 @@
 # -*- coding: utf-8 -*-
-##############################################################################
-#
-#    OpenERP, Open Source Management Solution
-#    Copyright (C) 2004-2010 Tiny SPRL (<http://tiny.be>).
-#
-#    This program is free software: you can redistribute it and/or modify
-#    it under the terms of the GNU Affero General Public License as
-#    published by the Free Software Foundation, either version 3 of the
-#    License, or (at your option) any later version.
-#
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
-
-from openerp import api
-from openerp.osv import fields, osv
+
+from openerp import api, fields, models
 from lxml import etree
 from openerp.tools.translate import _
-from openerp.exceptions import UserError
+from openerp.exceptions import UserError, ValidationError
 
-class followup(osv.osv):
+class Followup(models.Model):
     _name = 'account_followup.followup'
     _description = 'Account Follow-up'
     _rec_name = 'name'
-    _columns = {
-        'followup_line': fields.one2many('account_followup.followup.line', 'followup_id', 'Follow-up', copy=True),
-        'company_id': fields.many2one('res.company', 'Company', required=True),
-        'name': fields.related('company_id', 'name', string = "Name", readonly=True, type="char"),
-    }
-    _defaults = {
-        'company_id': lambda s, cr, uid, c: s.pool.get('res.company')._company_default_get(cr, uid, 'account_followup.followup', context=c),
-    }
-    _sql_constraints = [('company_uniq', 'unique(company_id)', 'Only one follow-up per company is allowed')] 
+
+    followup_line = fields.One2many('account_followup.followup.line', 'followup_id', 'Follow-up', copy=True)
+    company_id = fields.Many2one('res.company', 'Company', default=lambda self: self.env['res.company']._company_default_get('account_followup.followup'), required=True)
+    name = fields.Char(readonly=True)
```

Comment:
Why related field is converted to only Char ???
Use 'related' attribute to define a related field.
related: The complete sequence of fields to traverse is specified by the attribute.